service-locator/core/src/main/scala/com/lightbend/lagom/internal/client/ServiceNameMapper.scala

```diff
@@ -17,95 +17,56 @@ import scala.collection.JavaConverters._
 private[lagom] class ServiceNameMapper(config: Config) {
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  private val serviceLookupRegex = Some(config.getString("service-lookup-regex"))
-    .filter(_.nonEmpty)
-    .map(Pattern.compile)
-
   private val defaultPortName = Some(config.getString("defaults.port-name")).filter(_.nonEmpty)
   private val defaultPortProtocol = Some(config.getString("defaults.port-protocol")).filter(_.nonEmpty)
-  private val serviceNameSuffix = config.getString("service-name-suffix")
+  private val defaultScheme = Some(config.getString("defaults.scheme")).filter(_.nonEmpty)
 
-  private val portNameSchemeMapping = {
-    val mappings = config.getConfig("port-name-scheme-mapping")
+  private val serviceLookupMapping: Map[String, ServiceLookup] =
     config
-      .getObject("port-name-scheme-mapping")
+      .getObject("service-name-mappings")
+      .entrySet()
       .asScala
-      .map {
-        case (key, _) => key -> mappings.getString(key)
+      .map { entry =>
+        if (entry.getValue.valueType != ConfigValueType.OBJECT) {
+          throw new IllegalArgumentException(
```

Comment:
Out of scope here, but we can later report errors the same way Play does:

https://github.com/playframework/playframework/blob/b2143b6c7d253d4aa0a9f71ab7c98e4a039137b9/framework/src/play/src/main/scala/play/api/Configuration.scala#L149-L165